### PR TITLE
Reader follow button is now a custom view

### PR DESCRIPTION
**Pull request by nbradbury**
Feature-wise there's nothing new in this PR - all it does is add a custom view for the Reader follow button to simplify re-use. A big reason for this PR is the upcoming redesign will require changes to the follow button which will be (much) easier to deal with now the the button is a self-contained widget.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderFollowButton.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        inflate(context, R.layout.reader_follow_button, this);
+        mTextFollow = (TextView) findViewById(R.id.text_follow_button);
+    }
+
+    private void updateFollowText() {
+        mTextFollow.setSelected(mIsFollowed);
+        mTextFollow.setText(mIsFollowed ? R.string.reader_btn_unfollow : R.string.reader_btn_follow);
+    }
+
+    public void setIsFollowed(boolean isFollowed, boolean animateChanges) {
```
Comment (roundhill) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderFollowButton.java]: Doesn't really matter, but the method name with the two boolean parameters were a bit hard to understand when I saw it used in the code. Maybe `setIsFollowedAnimated` or something would help define the parameters?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderFollowButton.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                @Override
+                public void onAnimationRepeat(Animator animation) {
+                    updateFollowText();
+                }
+            });
+
+            AnimatorSet set = new AnimatorSet();
+            set.play(anim);
+            set.setDuration(200);
+            set.setInterpolator(new AccelerateDecelerateInterpolator());
```
Comment (roundhill) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderFollowButton.java]: Nice little animation :+1:

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderFollowButton.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderFollowButton.java]: You're right - that's bad style on my part, and I even made a similar comment on a recent PR from Danilo. I'll correct this.

**Comment by nbradbury** [comment]
@roundhill Ready for a re-review

**Comment by roundhill** [comment]
:shipit:

**Pull request merged**